helper/schema/schema.go

```diff
@@ -1452,19 +1469,19 @@ func (m schemaMap) validate(
 	err := validateExactlyOneAttribute(k, schema, c)
 	if err != nil {
 		return append(diags, diag.Diagnostic{
-			Severity:      diag.Error,
+			Severity:      diag.Severity(schema.ConditionsMode),
```

Comment:
Sorry to nitpick, but I'd feel much better about explicitly converting these rather than type casting, because if either value ever changes, I think the results would be surprising.

Maybe we could have a `SchemaConditionsMode.AsDiagSeverity` method that returned the `diag.Severity` for that mode? The method could be as simple as:

```go
func (s SchemaConditionsMode) AsDiagSeverity() diag.Severity {
  switch s {
    case SchemaConditionsModeWarning:
      return diag.Warning
    default:
      return diag.Error
  }
}
```

for example.